package/kedro_viz/models/graph.py

```diff
@@ -455,6 +468,11 @@ def __post_init__(self):
         )
 
 
+def create_metrics_plot(df: DataFrame) -> Dict[str, any]:
+            df = df.reset_index().rename(columns={'index': 'version'})
+            df = pd.melt(df, id_vars='version', var_name='metrics').sort_values(by='version')
+            return json.loads(pio.to_json(px.line(df, x="version", y='value', color='metrics')))

```

Comment:
I don't know much about plots, so this mostly looks like magic to me 😁 
But one thing I would suggest is rather than mutating your input `df` is to create new variables. It's best practice to not mutate your input, because it can cause nasty bugs that are hard to trace and explain. So here you could do something like:
```
renamed_columns_df = ...
melted_and_sorted_df = ...
return
```